### PR TITLE
refactor(datatype): deprecate `array`

### LINKED ISSUE #1851: Deprecate datatype.json() and datatype.array() for removal
### Clear and concise description of the problem

The methods return data that are most likely useless for the users as it contains arbritary data.

### Suggested solution

Deprecate the method and recommend helpers.multiple (for datatype.array) and a self built method for datatype.json.

### Alternative

_No response_

### Additional context

We decided that we don't want to return random "dynamic" objects from our API unless the user explicitly passed the inputs for the method: https://github.com/faker-js/faker/issues/1733#issuecomment-1433439926

### PR DESCRIPTION
**Pull request by xDivisionByZerox**
Part of #1851.

### CODE REVIEW

File: src/modules/datatype/index.ts [hunk 1]
```diff
@@ -425,6 +427,13 @@ export class DatatypeModule {
           max: number;
         } = 10
   ): Array<string | number> {
+    deprecated({
+      deprecated: 'faker.datatype.array()',
+      proposed: 'your own function to build complex arrays',
```
Comment (ST-DDT) [inline src/modules/datatype/index.ts:441]: IMO faker.helpers.multiple would work here better.

Comment (xDivisionByZerox) [reply src/modules/datatype/index.ts:441]: I think `helpers.multiple` and `datatype.array` are used for different use cases. `multiple` is for values of the same kind (multiple names, multiple hex numbers, etc.) while `array` gave you different values per element (well eighter `string` or `number`). Sure that's easy to achieve with `mutiple`, but its not a drop-in replacement which I thought the `proposed` property described.

There should probably be a "bigger" section in the migration guide on how to use `multiple` to mimic this function but for the warning in the code itself it doesn't fit.

At least all of that is how I see this.

Comment (ST-DDT) [reply src/modules/datatype/index.ts:441]: > There should probably be a "bigger" section in the migration guide on how to use `multiple` to mimic this function but for the warning in the code itself it doesn't fit.

Then you can add it as part of this PR. So we don't forget about it.

File: docs/guide/upgrading.md [hunk 2]
```diff
@@ -124,6 +124,11 @@ faker.number.float({ max: 100, precision: 0.01 }); // 35.21
 | `faker.datatype.float`  | `faker.number.float`                       |
 | `faker.datatype.bigInt` | `faker.number.bigInt`                      |
 
+### Deprecation of `datatype.array`
+
+The method `faker.datatype.array` has been deprecated and will be removed in v9.
+If you need an array of element you are better off creating your own one using `helpers.multiple`.
```
Comment (ST-DDT) [inline docs/guide/upgrading.md]:
```suggestion
If you need an array of useful values, you are better off creating your own one using `helpers.multiple`.
```

**Review by ST-DDT — DISMISSED** [review]

**Review by Shinigami92 — DISMISSED** [review]

File: docs/guide/upgrading.md [hunk 3]
```diff
@@ -124,6 +124,11 @@ faker.number.float({ max: 100, precision: 0.01 }); // 35.21
 | `faker.datatype.float`  | `faker.number.float`                       |
 | `faker.datatype.bigInt` | `faker.number.bigInt`                      |
 
+### Deprecation of `datatype.array`
+
+The method `faker.datatype.array` has been deprecated and will be removed in v9.
+If you need an array of useful values, you are better off creating your own one using `helpers.multiple`.
```
Comment (matthewmayer) [inline docs/guide/upgrading.md]: Using `faker.helpers.multiple`

**Review by matthewmayer — DISMISSED** [review]

**Review by ST-DDT — APPROVED** [review]

**Review by Shinigami92 — APPROVED** [review]

**Review by ejcheng — APPROVED** [review]

**Review by Shinigami92 — APPROVED** [review]

**Pull request merged**